contrib/README.md

```diff
@@ -0,0 +1,368 @@
+this on your end, please don't forget to update the queries accordingly.
+
+#### Register a function
+
+Define the function registration payload, using **AWS** as an example:
+
+```bash
+cat > function.json <<EOF
+{
+    "functionId": "echo",
+    "type": "awslambda",
+    "provider": {
+        "arn": "arn:aws:lambda:us-east-1:123456789012:function:event-gateway-tests-dev-echo",
+        "region": "us-east-1",
+        "awsAccessKeyID": "AAAAAAAAAAAAAAAAAAAA",
+        "awsSecretAccessKey": "AAAAaBcDeFgHiJqLmNoPqRsTuVwXyz0123456789"
+    }
+}
+EOF
+```
+
+Then call the registration endpoint with your json payload:
+
+```bash
+curl --request POST \
+  --url http://${EVENT_GATEWAY_URL}/v1/spaces/default/functions \
+  --header 'content-type: application/json' \
+  --header 'host: eventgateway.minikube' \
+  --data @function.json
+```
+
+And the corresponsing reply (if successful) should read something like the following:
+
+```bash
+{
+	"space": "default",
+	"functionId": "echo",
+	"type": "awslambda",
+	"provider": {
+		"arn": "arn:aws:lambda:us-east-1:123456789012:function:event-gateway-tests-dev-echo",
+		"region": "us-east-1",
+		"awsAccessKeyId": "AAAAAAAAAAAAAAAAAAAA",
+		"awsSecretAccessKey": "AAAAaBcDeFgHiJkLmNoPqRsTuVwXyZ0123456789"
+	}
+}
+```
+
+**NOTE:** if you try to overwrite an existing function, you will receive an error! To replace an existing function
+you will have to delete it first, then register the function once more. For example, trying to re-register the `echo`
+function will yield:
+
+```bash
+curl --request POST \
+   --url http://${EVENT_GATEWAY_URL}/v1/spaces/default/functions \
+   --header 'content-type: application/json' \
+   --header 'host: eventgateway.minikube' \
+   --data @function.json
+
+{
+    "errors": [{
+        "message": "Function \"echo\" already registered."
+    }]
+}
+```
+
+#### Query all functions
+
+To check for registered functions, query the `config` API with the `GET` request:
+
+```bash
+curl --request GET \
+  --url http://${EVENT_GATEWAY_URL}/v1/spaces/default/functions \
+  --header 'content-type: application/json' \
+  --header 'host: eventgateway.minikube'  | jq
+```
+
+You should see the functions list return your defined set of functions across all vendors.
+
+```bash
+{
+  "functions": [
+    {
+      "space": "default",
+      "functionId": "echo",
+      "type": "awslambda",
+      "provider": {
+        "arn": "arn:aws:lambda:us-east-1:123456789012:function:event-gateway-tests-dev-echo",
+        "region": "us-east-1",
+        "awsAccessKeyId": "AAAAAAAAAAAAAAAAAAAA",
+        "awsSecretAccessKey": "AAAAaBcDeFgHiJkLmNoPqRsTuVwXyZ0123456789"
+      }
+    }
+  ]
+}
+```
+
+#### Register an event
+
+To register an event, make sure to `POST` the event name to the `event-gateway`.
+
+```bash
+curl --request POST \
+  --url http://eventgateway.minikube/v1/spaces/default/eventtypes \
+  --header 'content-type: application/json' \
+  --data '{ "name": "eventgateway.function.invoked" }'
```

Comment:
It might be good to do the default `http.request` Event Type. Then, after setting up a subscription, show the user how to emit an `http.request` event and get a response.

On the other hand, it could require a working function example, which would add length to the tutorial. 🤔 Open to thoughts here.

Reply:
Yep good point. Not worth including in this resource ATM.